common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -221,14 +222,41 @@ public void saveDefaultConfig() {
     private List<ConfigHolder> getConfigOptions(Configurable parent) {
         List<ConfigHolder> options = new ArrayList<>();
 
-        for (Field configField : FieldUtils.getFieldsWithAnnotation(parent.getClass(), Config.class)) {
-            Config metadata = configField.getAnnotation(Config.class);
+        Field[] annotatedConfigs = FieldUtils.getFieldsWithAnnotation(parent.getClass(), RegisterConfig.class);
+        for (Field field : annotatedConfigs) {
+            try {
+                Object fieldValue = FieldUtils.readField(field, parent, true);
+                if (!(fieldValue instanceof Config)) {
+                    throw new RuntimeException("A non-Config class was marked with @ConfigInfo annotation: " + field);
```

Comment:
Let's update the annotation names in this method

Reply:
Thanks for spotting. IntelliJ keeps asking "update all references?" and I keep answering "only code" since it usually messes things up otherwise... This time it would have been wise to accept its help.